bmorph/core/workflows.py

```diff
@@ -70,19 +85,23 @@ def apply_annual_blendmorph(raw_upstream_ts, raw_downstream_ts,
     """
     bc_multipliers = pd.Series([])
     bc_totals = pd.Series([])
-
-    bc_down_multipliers = pd.Series([])
-    bc_down_totals = pd.Series([])
-
-    bc_up_multipliers = pd.Series([])
-    bc_up_totals = pd.Series([])
     
     training_window = slice(*training_window)
     bmorph_window = slice(*bmorph_window)
     reference_window = slice(*reference_window)
     raw_ts_window = slice(pd.to_datetime(raw_upstream_ts.index.values[0]),
                           pd.to_datetime(raw_upstream_ts.index.values[-1]))
     
+    # Check if there is enough data input to run mdcdedcdfm for both upstream
+    # and downstream bmorphs. Boolean used here instead of later to make certain
+    # both upstream and downstream use the same method and to minimze checks within
+    # the for-loop
+    run_mdcd = False
+    if not (raw_upstream_y is None or raw_downstream_y is None or train_upstream_y is None
+           or train_downstream_y is None or truth_upstream_y is None or truth_downstream_y 
+           is None):
+        run_mdcd = True
```

Comment:
You might replace this with something more readable like:
```
y_varlist = [raw_upstream_y, train_upstream_y, truth_upstream_y,
             raw_downstream_y, train_downstream_y, truth_downstream_y]
if np.any(list(map(lambda x: x is not None, y_varlist))):
 ....
```
You'll have to double check if it needs to be `if not np.any(...):`, but to me this reads a bit clearer.